### DISCUSSION TITLE
one to many join (made one to one by 'eq') produces an unwanted 'sub' list within each object of the 'data' list.

### DISCUSSION THREAD

**Discussion by SgtErnestBilko** [supabase/supabase#27607] [Questions] [answered]
I have a 'profiles' table linked by foreign key to an 'events' table.  The relationship is one to many.

The following code successfully retrieves the data.  The second 'eq' statement (a date selector) effectively produces a one to one relationship.

```
import { createClient } from "@/utils/supabase/server"  

async function getData(): Promise<item[]> {
  const supabase = createClient()
  const { data, error} = await supabase
    .from('profiles')
    .select(`name, email, admin, active, events (playing, dining, buggy, date))`)
    .eq('active', true)
    .eq('events.date', '2024-01-01')

etc
```

Here is a single item from the list produced
```
item:  {
  name: 'Please supply name',
  email: 'anemail@blah.com',
  admin: false,
  active: true,
  events: [ { date: '2024-01-01', buggy: false, dining: true, playing: true } ]
}
```

but I want it to be this

```
item:  {
  name: 'Please supply name',
  email: 'anemail@blah.com',
  admin: false,
  active: true,
  buggy: false
 dining: true
 playing: true
}
```
Of course I can use a 'for' loop to extract the events list or  an SQL command , but is there a way of getting the supabase query to produce the intended result ?

**Comment by GaryAustin1** [comment, marked as answer]
There is a spread operator, but it only works for to-one relationships.  As your events is a to-many it won't work.
https://supabase.com/blog/postgrest-11-prerelease#spreading-related-tables

The only approaches will be to do a view or rpc call to a Postgres function returning a table.

**Reply by SgtErnestBilko** [reply]
Thank you.
